server/schedule/filter/filters.go

```diff
@@ -49,14 +49,32 @@ func SelectSourceStores(stores []*core.StoreInfo, filters []Filter, opt *config.
 	})
 }
 
+// SelectSourceStoresWithCollector selects stores that be selected as source store from the list.
+func SelectSourceStoresWithCollector(stores []*core.StoreInfo, filters []Filter, opt *config.PersistOptions, collector *plan.Collector) []*core.StoreInfo {
```

Comment:
How about making it an optional parameter?

Reply:
I want making collector be an optional parameter